vizro-core/src/vizro/models/_components/card.py

```diff
@@ -1,48 +1,94 @@
         text (str): Markdown string to create card title/text that should adhere to the CommonMark Spec.
         href (str): URL (relative or absolute) to navigate to. If not provided the Card serves as a text card
             only. Defaults to `""`.
+        figure (CapturedCallable): Card like object to be displayed. Defaults to `None`. For more information see
+            available card callables in [`vizro.cards`](vizro.cards).
 
     """
 
     type: Literal["card"] = "card"
-    text: str = Field(
-        ..., description="Markdown string to create card title/text that should adhere to the CommonMark Spec."
-    )
+    text: str = Field("", description="Markdown string to create card text that should adhere to the CommonMark Spec.")
     href: str = Field(
         "",
         description="URL (relative or absolute) to navigate to. If not provided the Card serves as a text card only.",
     )
+    figure: Optional[CapturedCallable] = Field(
+        None, import_path=vc, description="Dynamic card function to be visualized on dashboard."
+    )
+
+    # Component properties for actions and interactions
+    _output_component_property: str = PrivateAttr("children")
+
+    # Validators
+    @validator("figure")
+    def validate_figure(cls, figure, values):
+        if figure is None:
+            return (
+                vc.nav_card(text=values["text"], href=values["href"], data_frame=pd.DataFrame())
+                if values["href"]
+                else vc.text_card(text=values["text"], data_frame=pd.DataFrame())
+            )
+        return figure
+
+    _validate_callable_mode = _callable_mode_validator_factory("card")
+    _validate_callable = validator("figure", allow_reuse=True, always=True)(_process_callable_data_frame)
+
+    def __call__(self, **kwargs):
+        kwargs.setdefault("data_frame", data_manager[self["data_frame"]].load())
+        figure = self.figure(**kwargs) if self.figure else None
+        return figure
+
+    def __getitem__(self, arg_name: str):
+        # pydantic discriminated union validation seems to try Graph["type"], which throws an error unless we
+        # explicitly redirect it to the correct attribute.
+        if self.figure:
+            if arg_name == "type":
+                return self.type
+            return self.figure[arg_name]
 
     @_log_call
     def build(self):
-        text = dcc.Markdown(self.text, dangerously_allow_html=False, id=self.id)
-        card_content = (
-            dbc.NavLink(
-                text,
-                href=get_relative_path(self.href) if self.href.startswith("/") else self.href,
-            )
-            if self.href
-            else text
-        )
+        # LQ: Don't really like this - is there a better way to do this? I need to provide different classNames to the outer
+        # dbc.Card based on the CapturedCallable provided. This is a bit hacky.
+        function_name = self.figure._function.__name__
```

Comment:
Then I think I would prefer option 1), because otherwise, we do some unnecessary HTML nesting. The dbc.Card itself just wraps the content inside an html `div` that automatically gets the Bootstrap `className` `card` attached, so it gets automatically styled by our defaults. And we basically just want to append to that `className`.

Reply:
Yes, exactly :D